giskard/ml_worker/utils/cache.py

```diff
@@ -0,0 +1,103 @@
+from typing import Any, Dict, List, Optional
+
+import hashlib
+import logging
+from pickle import PicklingError
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SimpleCache:
+    """
+    Simple cache for storing and retrieving results. Uses the LRU algorithm.
+    """
+
+    def __init__(self, max_results: int = 128):
+        """
+        Initialize the cache with a maximum number of results.
+
+        Args:
+            max_results (int): The maximum number of results to store in the cache.
+        """
+        self._max_results = max_results
+        self._results: Optional[Dict[str, Any]] = None
+        self._keys: Optional[List[str]] = None
+
+    def content(self):
+        """
+        Returns the content of the cache.
+        """
+        return self._results, self._keys
+
+    def start(self, cache_content: Dict[str, Any], cache_keys: List[str]):
+        """
+        Initialize the cache with existing content and keys.
+
+        Args:
+            cache_content (dict): The content to populate the cache.
+            cache_keys (list): The keys associated with the cache content.
+        """
+        self._results = cache_content
+        self._keys = cache_keys
+
+    def _generate_key(self, obj):
+        return hashlib.md5(repr(obj).encode(encoding="utf-8")).hexdigest()
```

Comment:
let's use `xxhash` that we use already anyway because it's a lot faster

Reply:
Sure, will change